opentelemetry/src/exporter/trace/mod.rs

```diff
@@ -10,28 +11,25 @@ use http::Request;
 use serde::{Deserialize, Serialize};
 #[cfg(all(feature = "http", feature = "reqwest"))]
 use std::convert::TryInto;
-use std::error::Error;
-use std::fmt::{Debug, Display};
+use std::fmt::Debug;
+#[cfg(feature = "http")]
+use std::fmt::{Display, Formatter};
 use std::sync::Arc;
 use std::time::SystemTime;
 
 pub mod stdout;
 
 /// Describes the result of an export.
-pub type ExportResult = Result<(), Box<dyn std::error::Error + Send + Sync + 'static>>;
-
-/// Timed out when exporting spans to remote
-#[derive(Debug, Default)]
-pub struct ExportTimedOutError {}
+pub type ExportResult = Result<(), TraceError>;
 
-impl Display for ExportTimedOutError {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str("export timed out")
+/// Marker trait for errors returned by exporters
+pub trait ExportError: std::error::Error + Send + Sync + 'static {
+    /// The name of exporter that returned this error
+    fn exporter_name(&self) -> &'static str {
```

Comment:
Is there a reason that this has a default impl? are there unnamed exporters?

Reply:
It's more for the http client error from our HttpClient implementation, which is most like not realted to exporter itself. It's also difficult to get expoter name from httpclient.